src/Tests/dotnet-workload-install.Tests/GivenWorkloadManifestUpdater.cs

```diff
@@ -13,51 +13,109 @@
+            var oldEnvVarVal = Environment.GetEnvironmentVariable(intervalEnvVar);
+            try
             {
-                Directory.CreateDirectory(Path.Combine(installedManifestDir, manifest.ToString()));
-                File.WriteAllText(Path.Combine(installedManifestDir, manifest.ToString(), _manifestFileName), GetManifestContent(new ManifestVersion("1.0.0")));
+                Environment.SetEnvironmentVariable(intervalEnvVar, "0");
+                manifestUpdater.BackgroundUpdateAdvertisingManifestsAsync().Wait();
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(intervalEnvVar, oldEnvVarVal);
             }
 
-            var manifestDirs = installedManifests
-                .Select(manifest => Path.Combine(installedManifestDir, manifest.ToString(), _manifestFileName))
-                .ToArray();
-            var workloadManifestProvider = new MockManifestProvider(manifestDirs);
-            var nugetDownloader = new MockNuGetPackageDownloader(dotnetRoot, manifestDownload: true);
-            var manifestUpdater = new WorkloadManifestUpdater(_reporter, workloadManifestProvider, nugetDownloader, testDir, testDir);
-
-            manifestUpdater.UpdateAdvertisingManifestsAsync(true).Wait();
-            var expectedDownloadedPackages = installedManifests
-                .Select(id => ((PackageId, NuGetVersion, DirectoryPath?, PackageSourceLocation))(new PackageId($"{id}.manifest-{featureBand}"), null, null, null));
+            var expectedDownloadedPackages = _installedManifests
+                .Select(id => ((PackageId, NuGetVersion, DirectoryPath?, PackageSourceLocation))(new PackageId($"{id}.manifest-6.0.100"), null, null, null));
             nugetDownloader.DownloadCallParams.Should().BeEquivalentTo(expectedDownloadedPackages);
+            File.Exists(sentinalPath).Should().BeTrue();
+            File.GetLastAccessTime(sentinalPath).Should().BeAfter(createTime);
+        }
+
+        [Fact]
+        public void GivenAdvertisingManifestUpdateItDoesNotUpdateWhenNotDue()
+        {
+            (var manifestUpdater, var nugetDownloader, var testDir) = GetTestUpdater();
+
+            var sentinalPath = Path.Combine(testDir, ".dotnet", _manifestSentinalFileName);
+            File.Create(sentinalPath);
+            var createTime = DateTime.Now;
+
+            manifestUpdater.BackgroundUpdateAdvertisingManifestsAsync().Wait();
+            nugetDownloader.DownloadCallParams.Should().BeEmpty();
+            File.GetLastAccessTime(sentinalPath).Should().BeBefore(createTime);
+        }
+
+        [Fact]
+        public void GivenAdvertisingManifestUpdateItHonorsDisablingEnvVar()
+        {
+            (var manifestUpdater, var nugetDownloader, _) = GetTestUpdater();
+
+            var disableEnvVar = "DOTNET_CLI_WORKLOAD_UPDATE_NOTIFY_DISABLE";
+            var oldEnvVarVal = Environment.GetEnvironmentVariable(disableEnvVar);
```

Comment:
don't change global state in a test. Tests cannot run in parallel .Mock it out or just not test it.